### ISSUE TITLE
Настройка порта для БД

### ISSUE THREAD

**Issue by jasmanchik** [local-deploy/dl#52]
Сейчас нет возможности подключаться к БД например из IDE, так как нет проброса порта на локальную машину.

Предлагаю в кфг docker-compose-mysql.yaml сделать параметр "port" и подставлять туда значение из .env, соответственно у каждого проекта будет доступен свой порт для подключения и они смогут существовать параллельно. 
Еще лучше при запуске проекта (dl up) сделать проверку на доступность данного порта (который задаем в .env), и выкидывать ошибку, что этот порт уже занят другим контейнером

**Comment by varrcan** [comment]
На самом деле, порт 3306 пробрасывается, но только под IP контейнера.
То есть можно посмотреть IP адрес контейнера с базой данных (через команду dl ps, например) и потом этот IP прописать в подключении IDE.
Минус в таком подходе, IP будет меняться при каждом перезапуске контейнеров.

Я сейчас изучаю возможности traefik, умеет ли он проксировать запросы к бд, то есть хочу попробовать сделать локальный домен в качестве имени хоста к базе.
Более сложным решением является запуск собственного DNS, который разрешает имена контейнеров. Но этот вариант потребует установки пакета resolvconf, то есть будет запрос на root доступ, что противоречит концепции DL. 

![img181120228674](https://user-images.githubusercontent.com/23196073/202647740-2d1c1f61-a0c6-4eeb-b1f6-01f371a8d4a3.png)
![img181120223828](https://user-images.githubusercontent.com/23196073/202647916-e7487b88-654a-436e-acf8-73923814e110.png)

**Comment by Bezponiki** [comment]
Идей как сделать такое же подключение к wsl2 нет ?)

**Comment by jasmanchik** [comment]
Можно просто добавить в .env параметр типа MYSQL_LOCAL_PORT, и из него брать порт для локальной машины. Сейчас просто попробовал - получилось и решил написать
<img width="205" alt="image" src="https://user-images.githubusercontent.com/40543800/216326254-16b79cc4-ce44-4b99-9d03-24438e22aa5c.png">
<img width="614" alt="image" src="https://user-images.githubusercontent.com/40543800/216326284-c9f391d8-4703-45a0-909a-09245e14f681.png">

**Comment by varrcan** [comment]
> Можно просто добавить в .env параметр типа MYSQL_LOCAL_PORT

Да, только если не указать MYSQL_LOCAL_PORT, то он будет по умолчанию 3306, таким образом одновременно 2 проекта уже не поднять, порт будет занят.

**Comment by varrcan** [comment]
К сожалению, через traefik так сделать нельзя.. 
https://github.com/traefik/traefik/issues/5155

**Comment by jasmanchik** [comment]
А есть способ проверить порт на занятость? Допустим если порт занят, то подбирать рандомный и так, пока не будет подобран свободный и показывать в результативном сообщении на каком порту доступна БД) 
А если порт указан в .env, и если он занят, то выкидывать ошибку и останавливать запуск контейнера
